src/main/java/com/mercuryCyclists/procurement/listener/KafkaListeners.java

```diff
@@ -0,0 +1,74 @@
+package com.mercuryCyclists.procurement.listener;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import com.mercuryCyclists.procurement.entity.Supplier;
+import com.mercuryCyclists.procurement.repository.SupplierRepository;
+import com.mercuryCyclists.procurement.service.SupplierService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.kafka.annotation.KafkaListener;
+import org.springframework.stereotype.Component;
+import org.springframework.web.client.RestTemplate;
+
+import javax.transaction.Transactional;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Listener component for Kafka messaging
+ */
+@Transactional
+@Component
+public class KafkaListeners {
+
+    private final SupplierRepository supplierRepository;
+    private final SupplierService supplierService;
+
+    private static final String GETPARTSFROMPRODUCT = "http://localhost:8081/api/v1/product/{productId}/part";
+    private static final RestTemplate restTemplate = new RestTemplate();
+    private static final JsonParser jsonParser = new JsonParser();
+
+    @Autowired
+    public KafkaListeners(SupplierRepository supplierRepository, SupplierService supplierService) {
+        this.supplierRepository = supplierRepository;
+        this.supplierService = supplierService;
+    }
+
+    /**
+     * Processes sales pushed to the backorder topic
+     * Extracts the saleId and productId from sale
+     * gets parts from productId
+     * gets supplier from parts
+     * saves sale based on supplier provided
+     * @param data pulled from backorder topic
+     */
+    @KafkaListener(topics = "backorder", groupId = "backorder")
+    void backOrderListener(String data){
+        try{
+            System.out.printf("Kafka message received, %s%n", data);
```

Comment:
Do we need this? Doesn't something like this get printed on the broker's console?

Reply:
good point, I think I was just debugging.

Should be logged but we can just rm